### ISSUE TITLE
ELEKTRA RNN

### ISSUE THREAD

**Issue by rogarciago** [rogarciago/ANALISIS-Y-PREDICCION-DE-MERCADO#4]
import numpy as np
import sklearn.metrics as metrics
import sklearn.metrics as sm
from sklearn.metrics import mean_squared_error
from sklearn.metrics import accuracy_score, confusion_matrix
import pandas as pd
import plotly.graph_objs as go
from pandas_datareader import data, wb
from datetime import date
import cufflinks as cf
import yfinance as yf
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import tensorflow as tf
from sklearn.preprocessing import MinMaxScaler
from keras.models import Sequential
from keras.layers import Dense
from keras.layers import LSTM
from keras.layers import Dropout
import math

# In[2]:

start=pd.to_datetime('2013-01-01')
end=pd.to_datetime('2018-01-08')

# In[3]:

ELEKTRA = data.DataReader('ELEKTRA.MX', 'yahoo', start, end)

# In[4]:

ELEKTRA_processed=ELEKTRA.iloc[:, 3:4].values

# In[5]:

scaler = MinMaxScaler(feature_range = (0, 1))
ELEKTRA_training_scaled = scaler.fit_transform(ELEKTRA_processed)

# In[6]:

x_train = []
y_train = []
for i in range(60, 1259):
    x_train.append(ELEKTRA_training_scaled[i-60:i, 0])
    y_train.append(ELEKTRA_training_scaled[i, 0])
x_train, y_train = np.array(x_train), np.array(y_train)

# In[7]:

x_train = np.reshape(x_train, (x_train.shape[0], x_train.shape[1], 1))

# In[8]:

model = Sequential()
model.add(LSTM(units=200, return_sequences=True, activation='relu',input_shape=(x_train.shape[1], 1)))
model.add(Dropout(0.2))

model.add(LSTM(units=200, return_sequences=True))
model.add(Dropout(0.2))

model.add(LSTM(units=200, return_sequences=True))
model.add(Dropout(0.2))

model.add(LSTM(units=200, return_sequences=False))
model.add(Dropout(0.2))

# In[9]:

model.add(Dense(units = 1))
model.compile(optimizer = 'adam', loss = 'mean_squared_error')

# In[10]:

model.fit(x_train, y_train, epochs = 100, batch_size = 32)

# In[11]:

start=pd.to_datetime('2018-01-08')
end=pd.to_datetime('2018-01-16')
ELEKTRA2 = data.DataReader('ELEKTRA.MX', 'yahoo', start, end)

# In[13]:

ELEKTRA2_processed=ELEKTRA2.iloc[:, 3:4].values

# In[14]:

ELEKTRA_total = pd.concat((ELEKTRA['Close'], ELEKTRA2['Close']), axis=0)

# In[15]:

test_inputs = ELEKTRA_total[len(ELEKTRA_total) - len(ELEKTRA2) - 60:].values

# In[16]:

test_inputs = test_inputs.reshape(-1,1)
test_inputs = scaler.transform(test_inputs)

# In[17]:

test_features = []
for i in range(60, 67):
    test_features.append(test_inputs[i-60:i, 0])

# In[18]:

test_features = np.array(test_features)
test_features = np.reshape(test_features, (test_features.shape[0], test_features.shape[1], 1))

# In[19]:

predictions = model.predict(test_features)

# In[20]:

predictions = scaler.inverse_transform(predictions)

# In[21]:

plt.figure(figsize=(10,6))
plt.plot(ELEKTRA2_processed, color='blue', label='Actual Elektra Stock Price')
plt.plot(predictions , color='red', label='Predicted Elektra Stock Price')
plt.title('Elektra Stock Price Prediction')
plt.xlabel('Date')
plt.ylabel('Elektra Stock Price')
plt.legend()
plt.show()

# In[27]:

print("Mean absolute error =", round(sm.mean_absolute_error(ELEKTRA2_processed, predictions), 2)) 
print("Mean squared error =", round(sm.mean_squared_error(ELEKTRA2_processed, predictions), 2)) 
print("Median absolute error =", round(sm.median_absolute_error(ELEKTRA2_processed, predictions), 2)) 
print("Explain variance score =", round(sm.explained_variance_score(ELEKTRA2_processed, predictions), 2)) 
print("R2 score =", round(sm.r2_score(ELEKTRA2_processed, predictions), 2))

# In[28]:

start=pd.to_datetime('2018-01-08')
end=pd.to_datetime('2018-02-08')
ELEKTRA3 = data.DataReader('ELEKTRA.MX', 'yahoo', start, end)

# In[29]:

ELEKTRA3_processed=ELEKTRA3.iloc[:, 3:4].values

# In[30]:

ELEKTRA_total = pd.concat((ELEKTRA['Close'], ELEKTRA3['Close']), axis=0)

# In[31]:

test_inputs = ELEKTRA_total[len(ELEKTRA_total) - len(ELEKTRA3) - 60:].values

# In[32]:

test_inputs = test_inputs.reshape(-1,1)
test_inputs = scaler.transform(test_inputs)

# In[57]:

test_features = []
for i in range(60, 82):
    test_features.append(test_inputs[i-60:i, 0])

# In[58]:

test_features = np.array(test_features)
test_features = np.reshape(test_features, (test_features.shape[0], test_features.shape[1], 1))

# In[59]:

predictions = model.predict(test_features)

# In[60]:

predictions = scaler.inverse_transform(predictions)

# In[61]:

plt.figure(figsize=(10,6))
plt.plot(ELEKTRA3_processed, color='blue', label='Actual Elektra Stock Price')
plt.plot(predictions , color='red', label='Predicted Elektra Stock Price')
plt.title('Elektra Stock Price Prediction')
plt.xlabel('Date')
plt.ylabel('Elektra Stock Price')
plt.legend()
plt.show()

# In[62]:

print("Mean absolute error =", round(sm.mean_absolute_error(ELEKTRA3_processed, predictions), 2)) 
print("Mean squared error =", round(sm.mean_squared_error(ELEKTRA3_processed, predictions), 2)) 
print("Median absolute error =", round(sm.median_absolute_error(ELEKTRA3_processed, predictions), 2)) 
print("Explain variance score =", round(sm.explained_variance_score(ELEKTRA3_processed, predictions), 2)) 
print("R2 score =", round(sm.r2_score(ELEKTRA3_processed, predictions), 2))

# In[ ]:

# In[ ]:
